### ISSUE TITLE
[clangd] `PrerequisiteModulesTests` Generating Unexpected Error when Scanning for Dependencies on macOS

### ISSUE THREAD

**Issue by qiongsiwu** [llvm/llvm-project#124153]
`clangd`'s `PrerequisiteModulesTests` are passing on macOS, but produces unexpected diagnostic errors. This is reproducible using commit `bb6e94a05d15d289e3685c5599f0eb905dc46925`. 

For example
```
$ ./tools/clang/tools/extra/clangd/unittests/ClangdTests --gtest_filter="PrerequisiteModulesTests.*"                                                        
Note: Google Test filter = PrerequisiteModulesTests.*
[==========] Running 8 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 8 tests from PrerequisiteModulesTests
[ RUN      ] PrerequisiteModulesTests.NonModularTest
[       OK ] PrerequisiteModulesTests.NonModularTest (10 ms)
[ RUN      ] PrerequisiteModulesTests.ModuleWithoutDepTest
[       OK ] PrerequisiteModulesTests.ModuleWithoutDepTest (2 ms)
[ RUN      ] PrerequisiteModulesTests.ModuleWithDepTest
Scanning modules dependencies for foo.h failed: error: invalid argument '-std=c++20' not allowed with 'C'

Scanning modules dependencies for foo.h failed: error: invalid argument '-std=c++20' not allowed with 'C'

Ignored diagnostic. argument unused during compilation: '-c'
Built module N:Part to /var/folders/50/tcq20qrs2wj563qz76hymwlc0000gn/T/clangd/module_files/N-part.cppm2503448646646106135-79-eb-1a-03-e9-78/N-Part.pcm
Scanning modules dependencies for foo.h failed: error: invalid argument '-std=c++20' not allowed with 'C'

Scanning modules dependencies for foo.h failed: error: invalid argument '-std=c++20' not allowed with 'C'

Ignored diagnostic. argument unused during compilation: '-c'
Built module M to /var/folders/50/tcq20qrs2wj563qz76hymwlc0000gn/T/clangd/module_files/M.cppm5130802603315192334-a5-3f-5b-38-18-91/M.pcm
...
```

@ChuanqiXu9 pointed out that the errors should be fixed in `clangd` (https://github.com/llvm/llvm-project/pull/66462#issuecomment-2608685410) instead of updating the tests.

**Comment by ChuanqiXu9** [comment]
Are you able to try again with trunk?

**Comment by qiongsiwu** [comment]
Sorry about the delay. I just tried with ff4faaa660e1e9e844f8d6d15f55d810411d9642 and I am still running into the error. 
```
Scanning modules dependencies for foo.h failed: error: invalid argument '-std=c++20' not allowed with 'C'

Ignored diagnostic. argument unused during compilation: '-c'
Built module N:Part to /var/folders/md/sl1wtkzd5zjc5mtfjvg38rym0000gn/T/clangd/module_files/N-part.cppm7423886303560459538-de-91-fa-f3-a6-8b/N-Part.pcm
Ignored diagnostic. argument unused during compilation: '-c'
Built module M to /var/folders/md/sl1wtkzd5zjc5mtfjvg38rym0000gn/T/clangd/module_files/M.cppm17642611620768641015-9f-97-5f-8b-67-d6/M.pcm
```
